From 8dddd47c2005c54af6c41e1a0f306076c9bfb4b4 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Tue, 5 Dec 2023 11:12:38 -0600
Subject: [PATCH 13/14] solus: Allow default font to be set

Add an option to allow the default font to be set. Set it to ter-v32b by default

Signed-off-by: Reilly Brogan <reilly@reillybrogan.com>
---
 factory/templates/vconsole.conf.in | 1 +
 meson.build                        | 3 +++
 meson_options.txt                  | 2 ++
 src/vconsole/vconsole-setup.c      | 3 +++
 4 files changed, 9 insertions(+)

diff --git a/factory/templates/vconsole.conf.in b/factory/templates/vconsole.conf.in
index f682487695..05c2b8535c 100644
--- a/factory/templates/vconsole.conf.in
+++ b/factory/templates/vconsole.conf.in
@@ -1,3 +1,4 @@
 # This is the fallback vconsole configuration provided by systemd.
 
 #KEYMAP={{ SYSTEMD_DEFAULT_KEYMAP }}
+#FONT={{ SYSTEMD_DEFAULT_FONT }}
diff --git a/meson.build b/meson.build
index 269a46e3f3..c048339991 100644
--- a/meson.build
+++ b/meson.build
@@ -1020,6 +1020,9 @@ if default_keymap == ''
 endif
 conf.set_quoted('SYSTEMD_DEFAULT_KEYMAP', default_keymap)
 
+default_font = get_option('default-font')
+conf.set_quoted('SYSTEMD_DEFAULT_FONT', default_font)
+
 localegen_path = get_option('localegen-path')
 if localegen_path != ''
         conf.set_quoted('LOCALEGEN_PATH', localegen_path)
diff --git a/meson_options.txt b/meson_options.txt
index 1909323850..db61d51e06 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -312,6 +312,8 @@ option('default-locale', type : 'string', value : 'C.UTF-8',
        description : 'default locale used when /etc/locale.conf does not exist')
 option('nspawn-locale', type : 'string', value : 'C.UTF-8',
        description : 'default locale used by systemd-nspawn when executing commands in a container')
+option('default-font', type : 'string', value : 'ter-v32b',
+       description : 'default font used when populating /etc/vconsole.conf')
 option('default-keymap', type : 'string', value : 'us',
        description : 'default keymap used when populating /etc/vconsole.conf')
 option('localegen-path', type : 'string', value : '',
diff --git a/src/vconsole/vconsole-setup.c b/src/vconsole/vconsole-setup.c
index e4780cbcc9..3b71afec36 100644
--- a/src/vconsole/vconsole-setup.c
+++ b/src/vconsole/vconsole-setup.c
@@ -104,6 +104,9 @@ static const char* context_get_config(Context *c, VCMeta meta) {
         if (meta == VC_KEYMAP)
                 return isempty(c->config[VC_KEYMAP]) ? SYSTEMD_DEFAULT_KEYMAP : c->config[VC_KEYMAP];
 
+        if (meta == VC_FONT)
+                return isempty(c->config[VC_FONT]) ? SYSTEMD_DEFAULT_FONT : c->config[VC_FONT];
+
         return empty_to_null(c->config[meta]);
 }
 
-- 
2.43.0

